opennms-provision/opennms-provisiond/src/main/java/org/opennms/netmgt/provision/service/operations/SaveOrUpdateOperation.java

```diff
@@ -59,7 +62,13 @@
     protected SaveOrUpdateOperation(Integer nodeId, String foreignSource, String foreignId, String nodeLabel, String location, String building, String city, ProvisionService provisionService, String rescanExisting) {
         super(provisionService);
 
-        m_node = new OnmsNode(new OnmsMonitoringLocation(location, location), nodeLabel);
+        m_node = new OnmsNode();
+
+        m_node.setLocation(Strings.isNullOrEmpty(location) ?
+                new OnmsMonitoringLocation(MonitoringLocationDao.DEFAULT_MONITORING_LOCATION_ID, MonitoringLocationDao.DEFAULT_MONITORING_LOCATION_ID) :
```

Comment:
Can we not set it to `null` here instead? The `updateLocation` from `ProvisionService.insertNode()` or `ProvisionerService.updateNode()` should handle it accordingly then, shouldn't it?